geonotebook/kernel.py

```diff
@@ -365,17 +371,17 @@ def _add_layer(layer_name):
             params['zIndex'] = len(self.layers)
 
             cb = self._remote.add_wms_layer(layer.name, layer.vis_url, params)\
-                .then(_add_layer, self.rpc_error)
```

Comment:
Are you talking about the 3 calls to `add_*_layer` functions? If so, #48 refactors this method so it only calls `self._remote.add_layer` once with the type as an argument, see [here](https://github.com/OpenGeoscience/geonotebook/blob/c63f446eed0884eb75b5e40706a9af54b5fba293/geonotebook/kernel.py#L389).